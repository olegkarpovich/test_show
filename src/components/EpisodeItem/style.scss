#episode-general-information {
  width: 100%;
  display: flex;
  flex-direction: row;
  align-items: top;

  #episode-main-image {
    float: left;

    img {
      max-width: 100%;
    }
  }

  #episode-description {
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    text-align: left;
  }
}

@media(max-width: 768px) {
  .ep-block div,
  #episode-description {
    font-size: 12px;
  }
}

@media(max-width: 480px) {
  #episode-general-information {
    flex-direction: column !important;

    #episode-description {
      font-size: 10px;
      text-align: center !important;
    }

    #episode-description p{
      display: block;
      width: 95%;
    }
  }
}
