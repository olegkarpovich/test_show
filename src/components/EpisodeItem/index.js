import React from 'react'
import './style.scss';
import {BackButton} from "../Buttons";

function EpisodeItem (props) {
    const {episode} = props;
    return (
        <section id="episode-general-information">
            <BackButton />
            <aside id="episode-main-image">
                <figure>
                    {episode.image && <img src={episode.image.medium} alt={episode.name}/>}
                </figure>
            </aside>
            <article id="episode-description">
                <h3>{episode.name}</h3>
                <div dangerouslySetInnerHTML={{__html: episode.summary}}/>
            </article>
        </section>
    )
}

export default EpisodeItem;
