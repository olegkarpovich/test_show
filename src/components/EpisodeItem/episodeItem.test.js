import React from 'react';
import { render, act } from '@testing-library/react';
import EpisodeItem from './index';

const episode = {
    episode: {
        image: {
            medium: 'static.tvmaze.com/uploads/images/medium_landscape/53/132617.jpg'
        },
        name: 'Escape from Monster Island',
        summary: 'Bubbles wins two tickets to a concert and has to pick between Blossom and Buttercup to go with her. Meanwhile, the Mayor\'s plane goes down over Monster Island and it\'s up to the girls to rescue him.'
    }
};

const component = () => (
    <EpisodeItem {...episode}/>
);

describe('EpisodeItem component', () => {
    it('should render episode details page', () => {
        const { container } = render(component());

        expect(container).toMatchSnapshot();
    });
});