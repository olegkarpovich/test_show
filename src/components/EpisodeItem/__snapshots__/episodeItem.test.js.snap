// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodeItem component should render episode details page 1`] = `
<div>
  <section
    id="episode-general-information"
  >
    <button
      id="back-to-show-button"
    >
      Back
    </button>
    <aside
      id="episode-main-image"
    >
      <figure>
        <img
          alt="Escape from Monster Island"
          src="static.tvmaze.com/uploads/images/medium_landscape/53/132617.jpg"
        />
      </figure>
    </aside>
    <article
      id="episode-description"
    >
      <h3>
        Escape from Monster Island
      </h3>
      <div>
        Bubbles wins two tickets to a concert and has to pick between Blossom and Buttercup to go with her. Meanwhile, the Mayor's plane goes down over Monster Island and it's up to the girls to rescue him.
      </div>
    </article>
  </section>
</div>
`;
